src/System/Console/ANSI/Unix.hs

```diff
@@ -81,7 +83,9 @@ hSupportsANSIWithoutEmulation h =
 -- (See Common-Include.hs for Haddock documentation)
 getReportedCursorPosition = bracket (hGetEcho stdin) (hSetEcho stdin) $ \_ -> do
   hSetEcho stdin False   -- Turn echo off
-  get
+  -- If, unexpectedly, no data is available on the console input stream then
+  -- the timeout will prevent the getChar blocking
+  fromMaybe "" <$> timeout 500 get
```

Comment:
Note that the argument to `timeout` is in microseconds. 500 microseconds may be too short a timeout.

Reply:
Sorry, I misread the documentation. Now fixed.